specifications/wallet/ADP-957-collateral.md

```diff
@@ -0,0 +1,201 @@
+2.  If the user migrates a wallet with non-ada assets, these assets will be
+    bundled together with ada in the resulting wallet. Indeed, the migration
+    algorithm is designed to coalesce bundles together as much as possible, in
+    order to:
+
+    a. minimize the number of transactions required to perform a migration;
+    b. minimize the overall fee required; and
+    c. maximize the proportion of a UTxO set that can be successfully migrated.
+
+## Excessive collateral
+
+The distribution of a wallet's UTxO set may be such that all the pure-ada
+entries are far in excess of what would be required for collateral.
+
+In this case, the user might be reluctant to use any of their existing entries
+as collateral.
+
+## Potential solutions for these issues
+
+1.  Do nothing in addition to satisfying the requirements above.
+
+    If the user encounters a situation where their wallet's UTxO set does not
+    have enough collateral, or where the UTxO set only has pure-ada entries
+    that are too large, then they themselves will need to manually create a
+    rebalancing transaction that creates the appropriate collateral entries.
+
+2.  Adjust the wallet so that it tries to maintain, over time, a UTxO set that
+    has some proportion of entries suitable for inclusion as collateral inputs
+    in future transactions. This might involve the following adjustments:
+
+    a.  Adjust the migration algorithm so that it attempts to create a target
+        UTxO distribution with at least some entries suitable for use as
+        collateral inputs. This would probably require a relatively large
+        amount of development and testing effort to achieve, as the migration
+        algorithm is currently designed to do the opposite of this.
+
+    b.  Adjust the way we call the standard coin selection algorithm, so that
+        if the requested transaction would evolve the UTxO set to a state where
+        there are no entries suitable for use as collateral inputs, we
+        automatically append one or more additional pure-ada outputs (similar
+        to change outputs) to the transaction, to rectify this. This would
+        almost certainly add extra complexity to the coin selection algorithm.
+
+3.  On accepting a user request to make a selection that requires collateral:
+
+    a.  If the wallet detects that the UTxO set has **sufficient** collateral,
+        then return just one selection, with the outputs the user asked for.
+
+    b.  If the wallet detects that the UTxO set has **insufficient** collateral,
+        then return an ordered list with two selections:
+
+        1. Selection 1: a transaction that creates appropriate collateral.
+        2. Selection 2: a transaction corresponding to the user request.
+
+    c.  If the wallet detects that the UTxO set has **sufficient** collateral,
+        but that it's not possible to satisfy the `collateralPercentMaximum`
+        parameter, then return an ordered list with two selections:
```

Comment:
This section should be removed, as we no longer have a `collateralPercentMaximum` parameter.